docs/proposals/ose-rebase.md

```diff
@@ -0,0 +1,55 @@
+## Rebase OSE on Origin
+
+Today, code in OSE lands by fetching and merging Origin manually. OSE-specific changes
+are scattered in the repo and it's hard to reason about any conflicts. This doc proposes
+moving all the changes in OSE that diverge from Origin into a set of commits that will sit
+on top of the Origin code[0] and a Jenkins job that will be rebasing master on a daily basis.
+Rebases are going to continue to happen only for the master branch.
+
+There are two types of commits that need to be carried on top of Origin:
+1. long-term carries related to branding and release tooling (master and release branches)
+2. CVEs and other backported bugfixes (release branches only)
+
+The Origin repository is set as a remote `upstream` branch in OSE so in master we can
+simply rebase upstream/master. If any conflicts are hit, manual resolution will be required. In
+the future, we can extend the conflict resolution mechanism in Jenkins to send an e-mail to the
+author and committer of the carry commit. Before the actual rebase happens, the last commit
+produced by tito during the previous rebase should be squashed into the main tito commit (one
+of the carry commits).
+
+A nightly build of OSE should follow the steps below in order to rebase on top of Origin:
+```
+git clone git@github.com/openshift/ose.git
+pushd ose
+git remote add upstream git@github.com/openshift/origin.git
+git fetch upstream
+git checkout master
+# The following two commands assume the last two commits in git history are the main tito
+# diff and the last tito tag commit.
+git reset --soft HEAD~2
+git commit -m "[CARRY][BUILD] Specfile updates"
```

Comment:
`tito` looks at the current tag to generate the new tag -- so if `tito tag` sees nothing it will use version `0` or something